lib/mastermind/score_filter_ai.rb

```diff
@@ -23,8 +19,12 @@ def next_guess(feedback)
 
     attr_reader :initial_guess, :set_of_guesses, :current_guess, :code_score_calc 
 
-    def update_set_of_guesses(feedback)
-      @set_of_guesses = @set_of_guesses.select do |a_guess|
+    def initial_guess?(feedback)
+      feedback == ScoringFeedback.new(-1, -1)
+    end
+
+    def filter_set_of_guesses(set_of_guesses, feedback)
+      set_of_guesses.select do |a_guess|
         feedback == code_score_calc.score(current_guess, a_guess)
       end
     end
```

Comment:
I've updated this method so that it doesn't change the state from within and will take in a set of guesses and feedback and return the filtered set.